From 06b37b7dd98b8439efc8c990277938450241d5db Mon Sep 17 00:00:00 2001
From: John Greene <jogreene@redhat.com>
Date: Tue, 16 Dec 2014 19:31:31 -0500
Subject: [ethernet] ixgbe: avoid possible read_reg panic caused by late method binding

Message-id: <1418758291-4961-1-git-send-email-jogreene@redhat.com>
Patchwork-id: 102104
O-Subject: [RHEL7.1 PATCH v4 bz1145772 9/8] ixgbe: avoid possible read_reg panic caused by late method binding
Bugzilla: 1145772
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Dean Nelson <dnelson@redhat.com>

[redhat] This is advance copy of patch that will be submitted to net next
when merge window opens again. Hence, no upstream commit hash and
title may change.

BZ: 1145772
Brew: https://brewweb.devel.redhat.com/taskinfo?taskID=8371279
Tested: Issue found and fixed Intel X550 testing. See BZ for details
Upstream: not yet available

We always identify the PHY in our reset_hw path anyway so there is no need
to do it in get_invariants().  The reason I even noticed this is that for
new hardware (X550em) we don't assign some methods until later in probe and
calling phy.ops.read_reg could lead to a panic.

Signed-off-by: Don Skidmore <donald.c.skidmore@intel.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/ixgbe/ixgbe_x540.c b/drivers/net/ethernet/intel/ixgbe/ixgbe_x540.c
index b16268e..923eff6 100644
--- a/drivers/net/ethernet/intel/ixgbe/ixgbe_x540.c
+++ b/drivers/net/ethernet/intel/ixgbe/ixgbe_x540.c
@@ -55,9 +55,6 @@ s32 ixgbe_get_invariants_X540(struct ixgbe_hw *hw)
 {
  struct ixgbe_mac_info *mac = &hw->mac;
 
- /* Call PHY identify routine to get the phy type */
- ixgbe_identify_phy_generic(hw);
-
  mac->mcft_size = IXGBE_X540_MC_TBL_SIZE;
  mac->vft_size = IXGBE_X540_VFT_TBL_SIZE;
  mac->num_rar_entries = IXGBE_X540_RAR_ENTRIES;
-- 
1.7.1